core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -122,28 +208,44 @@ object JsonAST {
       case _ => false
     }
 
-    /** Return nth element from JSON.
-      * Meaningful only to JArray, JObject and JField. Returns JNothing for other types.
-      * <p>
-      * Example:<pre>
-      * JArray(JInt(1) :: JInt(2) :: Nil)(1) == JInt(2)
-      * </pre>
-      */
+    /**
+     * Return `n`th element from a `JArray` or `JObject` and return `JNothing` for any other kind of `JValue`.
```

Comment:
I guess if we're going to indicate `n`th, we should make the parameter name `n`, or make this say “`i`-th”. Also would suggest a dash there to be clear on the fact that it refers to the parameter, rather than just being the word “nth”.

Also may be worth clarifying:
 - What happens for a `JObject`.
 - What happens for non-`JArray`/`JObject` (presumably `JNothing`).